File: src/main/java/com/intellij/tapestry/intellij/view/nodes/FileNode.java
package com.intellij.tapestry.intellij.view.nodes;

import com.intellij.ide.projectView.PresentationData;
import com.intellij.openapi.module.Module;
import com.intellij.psi.PsiFile;

public class FileNode extends TapestryNode {

    public FileNode(PsiFile file, Module module) {
        super(module);

        init(file, new PresentationData(file.getName(), file.getName(), file.getFileType().getIcon(), null));
    }
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/intellij/view/nodes/FileNode.kt
package com.github.rar91279.plugin.tapestry.intellij.view.nodes

import com.intellij.ide.projectView.PresentationData
import com.intellij.openapi.module.Module
import com.intellij.psi.PsiFile

class FileNode(file: PsiFile, module: Module) : TapestryNode(module) {
    init {
        init(file, PresentationData(file.name, file.name, file.fileType.icon, null))
    }
}
